### PR TITLE
EEP: Core Library Dependencies

### PR DESCRIPTION
**Pull request by dmikurube**
This is a new "retrospective" EEP to explain the big-bang breaking changes through the Embulk v0.10 series for cleaning up library dependencies from the Embulk core.

Read the rendered version at:

https://github.com/embulk/embulk/blob/eep-core-library-dependencies/docs/eeps/draft-core-library-dependencies.md

### CODE REVIEW

**Comment by dmikurube** [comment]
Refined the explanation (English representation). It's ready.

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍

File: docs/eeps/draft-core-library-dependencies.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+ICU4J was unlikely to cause dependency conflicts between the Embulk core and plugins, since many plugins did not depend on ICU4J.
+
+While on the other hand, ICU4J is so huge. For example, `icu4j-54.1.1.jar` is ~10.6MB, `icu4j-57.2.jar` is ~10.8MB, and `icu4j-74.2.jar` is ~13.6MB. ICU4J occipied a quarter to a half in Embulk packages. We doubted that it was really worth having the entire ICU4J dependency in the Embulk core, since ICU4J was only used for character set detection.
+
+Finally, ICU4J has been removed (#2) from the Embulk core since [Embulk v0.10.45](https://github.com/embulk/embulk/releases/tag/v0.10.45). Instead, we have copied some classes for character set detection from ICU4J 57.2 into [`embulk-util-guess`](https://github.com/embulk/embulk-util-guess). They are licensed under the ICU License, which is considered compatible with the X license, hence the MIT License. Guess plugins that need to detect character sets would use their own character set detection in `embulk-util-guess`.
```
[the remark this answers is not in the document]
Comment (dmikurube) [reply docs/eeps/draft-core-library-dependencies.md:151]: Which is the same as the "X11 License":

https://en.wikipedia.org/wiki/MIT_License#X11_License

**Comment by dmikurube** [comment]
Thanks! Let's merge it as EEP-7 as `Final`.

**Pull request merged**